src/core/iam/identhandler/identmodules/fileidentifier/fileidentifier.cpp

```diff
@@ -114,8 +136,10 @@ Error FileIdentifier::ReadSubjects()
         return AOS_ERROR_WRAP(err);
     }
 
-    if (mSubjectsListener) {
-        mSubjectsListener->SubjectsChanged(mSubjects);
+    for (auto* subscriber : mSubscribers) {
```

Comment:
Do not notify subscribers in init. Usually, subscribers will call GetSubjects and then subscribe to changes.